File: backend/parking-domain/src/main/java/com/chatchatabc/parking/impl/infra/service/JedisServiceImpl.java
package com.chatchatabc.parking.impl.infra.service;

import com.chatchatabc.parking.infra.service.JedisService;
import org.springframework.beans.factory.annotation.Value;
import redis.clients.jedis.Jedis;

public class JedisServiceImpl implements JedisService {
    @Value("${server.jedis.host}")
    private String host;

    @Value("${server.jedis.port}")
    private String port;

    /**
     * Set key value to redis
     *
     * @param key   the key
     * @param value the value
     * @param ttl   the ttl
     */
    @Override
    public void set(String key, String value, Long ttl) {
        // TODO: Fix this with Jedis Pool
        Jedis jedis = new Jedis(host, Integer.parseInt(port));
        if (ttl != null) {
            jedis.setex(key, ttl, value);
        } else {
            jedis.set(key, value);
        }
        jedis.close();
    }

    /**
     * Get value from redis
     *
     * @param key the key
     * @return the value
     */
    @Override
    public String get(String key) {
        // TODO: Fix this with Jedis Pool
        Jedis jedis = new Jedis(host, Integer.parseInt(port));
        String value = jedis.get(key);
        jedis.close();
        return value;
    }

    /**
     * Delete key from redis
     *
     * @param key the key
     */
    @Override
    public void del(String key) {
        // TODO: Fix this with Jedis Pool
        Jedis jedis = new Jedis(host, Integer.parseInt(port));
        jedis.del(key);
        jedis.close();
    }
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/parking/impl/infra/service/JedisServiceImpl.kt
package com.chatchatabc.parking.impl.infra.service

import com.chatchatabc.parking.infra.service.JedisService
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import redis.clients.jedis.Jedis

@Service
class JedisServiceImpl (
    @Value("\${server.jedis.host}")
    private val host: String,

    @Value("\${server.jedis.port}")
    private val port: String,
) : JedisService {
    /**
     * Set key-value to Redis
     */
    override fun set(key: String, value: String, ttl: Long?) {
        // TODO: Fix this with Jedis Pool
        val jedis = Jedis(host, port.toInt())
        if (ttl != null) {
            jedis.setex(key, ttl, value)
        } else {
            jedis.set(key, value)
        }
        jedis.close()
    }

    /**
     * Get value from Redis
     */
    override fun get(key: String): String? {
        // TODO: Fix this with Jedis Pool
        val jedis = Jedis(host, port.toInt())
        val value = jedis.get(key)
        jedis.close()
        return value
    }

    /**
     * Delete key from Redis
     */
    override fun del(key: String) {
        // TODO: Fix this with Jedis Pool
        val jedis = Jedis(host, port.toInt())
        jedis.del(key)
        jedis.close()
    }
}